### ISSUE TITLE
bind, crt, ignore dot files in certificate directory

### ISSUE THREAD

**Issue by anvilvapre** [haproxy/haproxy#1689]
### Your Feature Request

ignore .(dot) files in certificate dirs. either as an option, by default, or via an include/exclude file extension filter.

i.e. `bind <ip> ssl crt <dir>`, where `dir` contains a `.gitignore` file. 

### What are you trying to do?

`
haproxy -c -f /etc/haproxy/haproxy.cfg
`

`[ALERT] 125/215704 (1782340) : parsing [/etc/haproxy/haproxy.cfg:115] : 'bind <ip>:443' : unable to load SSL private key from PEM file '/etc/haproxy/private/cert_dir/.gitignore'.
`

### Output of `haproxy -vv`

```plain
`
haproxy -vv                                                                                      
HA-Proxy version 2.0.13-2ubuntu0.5 2022/03/02 - https://haproxy.org/                                                      
Build options :                                                                                                           
  TARGET  = linux-glibc                                                                                                   
  CPU     = generic                                                                                                       
  CC      = gcc                                                                                                           
  CFLAGS  = -O2 -g -O2 -fdebug-prefix-map=/build/haproxy-qHmAxE/haproxy-2.0.13=. -fstack-protector-strong -Wformat -Werr  or=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fno-strict-aliasing -Wdeclaration-after-statement -fwrapv -Wno-addre  ss-of-packed-member -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-old-style-declaration -Wno-ignored-qu  alifiers -Wno-clobbered -Wno-missing-field-initializers -Wno-implicit-fallthrough -Wno-stringop-overflow -Wno-cast-funct  ion-type -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference                     
  OPTIONS = USE_PCRE2=1 USE_PCRE2_JIT=1 USE_REGPARM=1 USE_OPENSSL=1 USE_LUA=1 USE_ZLIB=1 USE_SYSTEMD=1                    

Feature list : +EPOLL -KQUEUE -MY_EPOLL -MY_SPLICE +NETFILTER -PCRE -PCRE_JIT +PCRE2 +PCRE2_JIT +POLL -PRIVATE_CACHE +TH  READ -PTHREAD_PSHARED +REGPARM -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H -VSYSCA  LL +GETADDRINFO +OPENSSL +LUA +FUTEX +ACCEPT4 -MY_ACCEPT4 +ZLIB -SLZ +CPU_AFFINITY +TFO +NS +DL +RT -DEVICEATLAS -51DEGR  EES -WURFL +SYSTEMD -OBSOLETE_LINKER +PRCTL +THREAD_DUMP -EVPORTS
```

**Comment by lukastribus** [comment]
Yeah, I suggest we should just ignore dot files. No new configuration directives, filters and other feature creep please.

**Comment by wlallemand** [comment]
I pushed this for "crt" and "ca-file". This will be available in 2.6. I don't think that will be backported in previous branches since it is a breaking change.

**Comment by anvilvapre** [comment]
Thank you.
